assets/src/dashboard/types.js

```diff
@@ -112,7 +116,18 @@ export const PageSizePropType = PropTypes.shape({
 export const StoryMenuPropType = PropTypes.shape({
   handleMenuToggle: PropTypes.func.isRequired,
   contextMenuId: PropTypes.number.isRequired,
-  handleMenuItemSelected: PropTypes.func.isRequired,
+  menuItemActions: PropTypes.shape({
+    default: PropTypes.func.isRequired,
+    [STORY_CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.PREVIEW]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.RENAME]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.CREATE_TEMPLATE]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.DELETE]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK]: PropTypes.func,
+    [STORY_CONTEXT_MENU_ACTIONS.CLOSE]: PropTypes.func,
+  }),
```

Comment:
This is annoying. it'd be cool if we could do 
```
  menuItemActions: PropTypes.shape({
    default: PropTypes.func.isRequired,
    [PropTypes.oneOf(STORY_CONTEXT_MENU_ACTIONS)]: PropTypes.func,
  }),
```

but sadly it does not work like that with proptypes. Typescript would be beneficial here :sweat_smile:

Reply:
can we do 
```
menuItemActions: PropTypes.shape({
    default: PropTypes.func.isRequired,
    ...Object.keys(STORY_CONTEXT_MENU_ACTIONS).reduce(
      (menuItemActionsShape, menuActionKey) => ({
        ...menuItemActionsShape,
        [menuActionKey]: PropTypes.func,
      }),
      {}
    ),
  }),
```

Reply:
genius!

Reply:
Moved to #6516